src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectivityTests/ConnectivityTest.cs

```diff
@@ -368,5 +371,76 @@ public static void ConnectionOpenDisableRetry()
             duration = timer.Elapsed;
             Assert.True(duration.Seconds > 5, $"Connection Open() with retries took less time than expected. Expect > 5 sec with transient fault handling. Took {duration.Seconds} sec.");                //    sqlConnection.Open();
         }
+
+        private const string ConnectToPath = "SOFTWARE\\Microsoft\\MSSQLServer\\Client\\ConnectTo";
+        private static bool CanCreateAliases()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ||
+                !DataTestUtility.IsTCPConnStringSetup())
+            {
+                return false;
+            }
+
+            using (WindowsIdentity identity = WindowsIdentity.GetCurrent())
+            {
+                WindowsPrincipal principal = new(identity);
+                if (!principal.IsInRole(WindowsBuiltInRole.Administrator))
+                {
+                    return false;
+                }
+            }
+
+            using RegistryKey key = Registry.LocalMachine.OpenSubKey(ConnectToPath, true);
+            if (key == null)
+            {
+                // Registry not writable
+                return false;
+            }
+
+            SqlConnectionStringBuilder b = new(DataTestUtility.TCPConnectionString);
+            if (!DataTestUtility.ParseDataSource(b.DataSource, out string hostname, out int port, out string instanceName) ||
+                !string.IsNullOrEmpty(instanceName))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [ConditionalFact(nameof(CanCreateAliases))]
+        public static void ConnectionAliasTest()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                throw new Exception("Alias test only valid on Windows");
+            }
+
+            if (!CanCreateAliases())
+            {
+                throw new Exception("Unable to create aliases in this environment. Windows + Admin + non-instance data source required.");
+            }
```

Comment:
I'm wondering how the conditional fact lets the test run when I don't have the registry key! And the test throws the "Unable to create aliases in this environment. Windows + Admin + non-instance data source required." exception, whilst it's passed one time.

Reply:
Yes. This is what I meant by the ConditionalFact being inconsistent across different environments. I remember I spent quite some time trying to figure out why it was not working the way it should and landed on this solution.